vortex-ffi/src/duckdb/mod.rs

```diff
@@ -30,11 +29,46 @@ pub unsafe extern "C-unwind" fn vx_dtype_to_duckdb_logical_type(
 ) -> duckdb_logical_type {
     let dtype = unsafe { dtype.as_ref().vortex_expect("null dtype") };
 
-    try_or(error, ptr::null_mut(), || {
+    try_or(error, null_mut(), || {
         Ok(dtype.to_duckdb_type()?.into_owning_ptr())
     })
 }
 
+/// Converts a DuckDB type into a vortex type
+#[unsafe(no_mangle)]
+pub unsafe extern "C-unwind" fn vx_duckdb_logical_type_to_dtype(
+    type_array: *const duckdb_logical_type,
+    nullable: *const c_uchar,
+    names: *const *const c_char,
```

Comment:
Why does this function take names, len, etc? Aren't these inside duckdb_logical_type?

Reply:
Nope columns are unname

Reply:
Perhaps name the function `vx_duckdb_schema_to_dtype`? Where schema then consists of `column_types: *const duckdb_logical_type, column_names: *const *const c_char, column_count: ...`